resources/aws/elb.go

```diff
@@ -225,7 +225,7 @@ func (lb ELB) findExisting() (*elb.LoadBalancerDescription, error) {
 	descriptions := resp.LoadBalancerDescriptions
 
 	if len(descriptions) == 0 {
-		return nil, NamedResourceNotFoundError{Name: lb.Name}
+		return nil, NotFoundError{ResourceType: ELBType, Name: lb.Name}
```

Comment:
Why are we doing this? When you only want to transport additional information for the message you can simply annotate the error with `MaskAnyf`. Lets make it not more complicated than it is. Lets just do something like this. 

```
return microerror.MaskAnyf(notFoundError, "%s with name %s not found", ELBType, lb.Name)